### PR TITLE
Add `await_each`

### PR DESCRIPTION
**Pull request by mristin**
* Added `await_each` in asynctools

* Documented `await_each`

* Fixes the issue #28

### CODE REVIEW

**Comment by mristin** [comment]
@maxfischer2781 this is now review-ready.

File: asyncstdlib/asynctools.py [hunk 1]
```diff
@@ -174,3 +177,31 @@ async def head_tail(iterable, leading=5, trailing=5):
...
+    .. code-block:: python3
+
+        import asyncstdlib as a
+
+         async def async_is_non_negative(x: int) -> bool:
+              return x > 0
+
+         all_non_negative = await a.all(
+             a.await_each(
+                 async_is_non_negative(x) for x in [1, 2, 3]))
```
Comment (maxfischer2781) [inline asyncstdlib/asynctools.py:204]: I'm somewhat unhappy with this example. It's not really clear how "lack of lambda functions" relates to the code.
It is perfectly possible to do
```python3
await a.all(await async_is_non_negative(x) for x in [1, 2, 3])
await a.all(a.map(await async_is_non_negative, [1, 2, 3]))
```

Perhaps just implying there are some pre-built awaitables would do the trick:
```python3
okay = await a.all(a.await_each([check_a, check_b, check_c]))
```

File: docs/source/api/asynctools.rst [hunk 2]
```diff
@@ -19,3 +19,5 @@ Iterator lifetime
 
 .. autofunction:: scoped_iter(iterable: (async) iter T)
     :async-with: :async iter T
+
+.. autofunction:: await_each(awaitables: Iterable[Awaitable[T]]) -> AsyncIterable[T]
```
Comment (maxfischer2781) [inline docs/source/api/asynctools.rst:23]:
```suggestion
.. autofunction:: await_each(awaitables: iter await T)
    :async-for: :T
```

**Review by maxfischer2781 — CHANGES_REQUESTED** [review]
Looks almost good to go. Please check out the example in the docstring.

File: asyncstdlib/asynctools.py (hunk 1 above)
Comment (mristin) [reply asyncstdlib/asynctools.py:204]: I re-read and you are right, this is a confusing example. Would putting icontract be too much for the example? Maybe you have a better idea? Any other library or operation?

Comment (mristin) [reply asyncstdlib/asynctools.py:204]: Actually, I'll leave out mentioning the lack of lambda functions and just give your example with the checks, where checks are functions returning booleans.

**Comment by maxfischer2781** [comment]
LGTM. Do you want to make any more changes @mristin?

**Review by maxfischer2781 — APPROVED** [review]
LGTM. Do you want to make any more changes @mristin?

**Comment by mristin** [comment]
@maxfischer2781 no more changes from my side, it's ready for squash & merge.

**Pull request merged**